compiler/javac/src/org/jetbrains/kotlin/javac/Javac.kt

```diff
@@ -0,0 +1,227 @@
+                                                  .map { it.value }
+
+    fun findClassesFromPackage(fqName: FqName) = javaClasses.filter { it.key?.parentOrNull() == fqName }
+                                                         .flatMap { it.value.withInnerClasses() } +
+                                                 elements.getPackageElement(fqName.asString())
+                                                         ?.members()
+                                                         ?.elements
+                                                         ?.filterIsInstance(TypeElement::class.java)
+                                                         ?.map { JavacClass(it, this) }
+                                                         .orEmpty()
+
+    fun getTreePath(tree: JCTree, compilationUnit: CompilationUnitTree): TreePath = trees.getPath(compilationUnit, tree)
+
+    private inline fun <reified T> Iterable<T>.toJavacList() = JavacList.from(this)
+
+    private fun findClassInSymbols(fqName: String) = elements.getTypeElement(fqName)?.let { JavacClass(it, this) }
+
+    private fun findPackageInSymbols(fqName: String) = elements.getPackageElement(fqName)?.let { JavacPackage(it, this) }
+
+    private fun JavacFileManager.setClassPathForCompilation() = apply {
+        setLocation(StandardLocation.CLASS_PATH,
+                    getLocation(StandardLocation.CLASS_PATH) + getLocation(StandardLocation.CLASS_OUTPUT))
+
+        val reader = ClassReader.instance(context)
+        val names = Names.instance(context)
+        val outDirName = getLocation(StandardLocation.CLASS_OUTPUT).firstOrNull()?.path ?: ""
+
+        list(StandardLocation.CLASS_OUTPUT, "", setOf(JavaFileObject.Kind.CLASS), true)
+                .forEach {
+                    val fqName = it.name
+                            .substringAfter(outDirName)
+                            .substringBefore(".class")
+                            .replace("/", ".")
+                            .let { if (it.startsWith(".")) it.substring(1) else it }
+                            .let(names::fromString)
+
+                    symbols.classes[fqName]?.let { symbols.classes[fqName] = null }
+                    val symbol = reader.enterClass(fqName, it)
+
+                    (elements.getPackageOf(symbol) as? Symbol.PackageSymbol)?.let {
+                        it.members_field.enter(symbol)
+                        it.flags_field = it.flags_field or Flags.EXISTS.toLong()
+                    }
+                }
+
+    }
+
+    private fun JavaClass.withInnerClasses(): List<JavaClass> = listOf(this) + innerClasses.flatMap { it.withInnerClasses() }
+
+}
+
+private object AnyJavaSourceVirtualFile : VirtualFile() {
+    override fun refresh(asynchronous: Boolean, recursive: Boolean, postRunnable: Runnable?) {}
+
+    override fun getLength() = 0L
+
+    override fun getFileSystem() = throw UnsupportedOperationException()
```

Comment:
May be (if you need this object at all) it's better to provide some message like `"Should never be called"`.